File: src (cpp)/animal/mammalia/giraffe.cpp
#include "giraffe.h"

Giraffe::Giraffe(int _x, int _y, int _weight) : def_ID(3),
                                                def_ratio_meat(0),
                                                def_ratio_plant(60) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = false;
  weight = _weight;
}
void Giraffe::Interact() {
  std::cout << "The giraffe is looking down to you, literally" << std::endl;
}

File: src/virtualzoo/animal/mammalia/Giraffe.java
package virtualzoo.animal.mammalia;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Herbivora;

/**
 * Kelas Giraffe mendefinisikan atribut untuk giraffe.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class Giraffe extends Mammalia {

  /** <p>
   * Constructor
   *
   * Menciptakan giraffe dengan posisi (locX,locY) dan berat animalWeight.
   * </p>
   * @param locX locX Nilai absis posisi.
   * @param locY locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat giraffe.
   */
  public Giraffe(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Herbivora(animalWeight, 60);
    id = 6;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan giraffe.
   */
  public void interact() {
    System.out.println("The giraffe is looking down to you, literally");
  }
}
